testsuite/openshift/metrics/prometheus.py

```diff
@@ -41,21 +43,21 @@ def values(self) -> list[float]:
         return [float(m["value"][1]) for m in self.metrics]
 
 
-class Prometheus:
+class Prometheus(LifecycleObject):
     """Interface to the OpenShift Prometheus client"""
 
     def __init__(self, url: str, token: str, namespace: str = None):
         self.token = token
+        self.url = url
```

Comment:
```suggestion
class Prometheus(LifecycleObject):
    """Interface to the OpenShift Prometheus client"""
    
    _client: KuadrantClient
    client: ApyProxy

    def __init__(self, url: str, token: str, namespace: str = None):
        self.token = token
        self.url = url
```
maybe something like this would work for variable declaration and to avoid the `type: ignore` thing. But you'd probably need to check `hasattr(self, "_client")` before closing the connection instead. I'm fine with both variants

Reply:
`_client` and `client` would be class level variables and would be shared between instances